### DISCUSSION TITLE
How to make PrimaryKeyRelatedField read date from field with `_id` suffix?

### DISCUSSION THREAD

**Discussion by heckad** [encode/django-rest-framework#7924] [Question & Answer] [answered]
I have a field with id to another model with the name `model_id`, and I want to access the model from the `model` field in validated_data dict, not the `model_id` field. How to do this?

**Comment by jerinpetergeorge** [comment, marked as answer]
Please add a brief context of your question, else, it would be difficult to suggest anything

**Reply by heckad** [reply]
For example, I have a serializer

```
Serializer(serializers.Serializer):
    some_model_field_id = serializers.PrimaryKeyRelatedField(
        queryset=SomeModel.objects.all(),
    )
```

How to make this code valid?
```
data = Serializer(data=request.data)
data.is_is_valid(raise_exception=True)

some_model = data.validated_data['some_model_field']  # access without _id suffix   
```

**Reply by jerinpetergeorge** [reply]
You might need something like this,

<pre><code>class SomeSerializer(serializers.Serializer):
    some_model_field_id = serializers.PrimaryKeyRelatedField(
        queryset=SomeModel.objects.all(),
    )

    <b>def validate(self, attrs):
        attrs = super().validate(attrs)
        attrs["some_model_field"] = attrs["some_model_field_id"]
        return attrs</b></code></pre>

But, I wouldn't override `validate()` ***"only for this"***

**Reply by heckad** [reply]
Is there any mechanism (something like field_name attr) that could indicate from which field to take data and which write?

**Reply by jerinpetergeorge** [reply]
> You might need something like this,
> 
> ```
> class SomeSerializer(serializers.Serializer):
>     some_model_field_id = serializers.PrimaryKeyRelatedField(
>         queryset=SomeModel.objects.all(),
>     )
> 
>     def validate(self, attrs):
>         attrs = super().validate(attrs)
>         attrs["some_model_field"] = attrs["some_model_field_id"]
>         return attrs
> ```
> 
> But, I wouldn't override `validate()` _**"only for this"**_



I think you can make use of the ***`source`*** argument as

```python
class SomeSerializer(serializers.Serializer):
    field_name_id = serializers.CharField(source="field_name")


s = SomeSerializer(data={"field_name_id": "foo-bar"})
s.is_valid(True)
print(s.validated_data) # OrderedDict([('field_name', 'foo-bar')])
print(s.data)  # {'field_name_id': 'foo-bar'}
```

**Reply by heckad** [reply]
This is the answer to the question. Thank you so much!

**Reply by lovelydinosaur** [reply]
Since the discussions interface is fairly new for a bunch of us, I figure I'll illuminate that there's a "mark as answer" control in the top right of the discussion. I'd *assumed* this is available to all(?) but perhaps it's only there for maintainers?

<img width="278" alt="Screenshot 2021-04-19 at 12 12 09" src="https://user-images.githubusercontent.com/647359/115227577-8d750980-a108-11eb-8ec9-12d95dd883c2.png">

**Reply by jerinpetergeorge** [reply]
>I'd assumed this is available to all(?) but perhaps it's only there for maintainers?

It's not available for me. I think it's exclusively for maintainers :disappointed:
